src/main/java/org/springframework/integration/dsl/context/IntegrationFlowContext.java

```diff
@@ -149,8 +149,11 @@ public void register(String flowId, IntegrationFlow integrationFlow, boolean aut
 	 */
 	public synchronized void remove(String flowId) {
 		if (this.registry.containsKey(flowId)) {
+			IntegrationFlow theFlow = this.registry.remove(flowId);
+			if (theFlow instanceof Lifecycle) {
+				((Lifecycle) theFlow).stop();
+			}
```

Comment:
Perhaps we should check for `SmartLifecyle` as well and only start/stop based on `isRunning()` ?

I wonder if that would be better - anyway - instead of a boolean argument - test for `SmartLifecycle` and `isRunning()` for consistency - seems odd to have a separate boolean and just use `Lifecycle`.